web/elm/src/Views/CommentBar.elm

```diff
@@ -0,0 +1,345 @@
+                        [ style "width" "60px"
+                        , style "display" "flex"
+                        , style "justify-content" "flex-end"
+                        ]
+                        (case model.state of
+                            Viewing _ ->
+                                [ editButton model state ]
+
+                            Editing _ ->
+                                [ saveButton model state ]
+
+                            Saving _ ->
+                                [ loadingButton ]
+                        )
+                    ]
+
+                else
+                    [ Html.pre
+                        Styles.commentBarText
+                        [ Html.text (getContent model) ]
+                    ]
+               )
+        )
+
+
+commentSetCallback : ( String, Result Http.Error () ) -> Model -> Model
+commentSetCallback ( content, result ) model =
+    { model
+        | state =
+            case ( result, model.state ) of
+                ( Ok (), Viewing _ ) ->
+                    Viewing content
+
+                ( Ok (), Editing state ) ->
+                    Editing { state | cached = content }
+
+                ( Ok (), Saving _ ) ->
+                    Viewing content
+
+                ( Err _, Saving state ) ->
+                    Editing state
+
+                ( _, state ) ->
+                    state
+    }
+
+
+update : Model -> UpdateState -> Message.Message -> ( Model, List Effect )
+update model updateState msg =
+    case msg of
+        Message.Click (Message.CommentBarButton kind id) ->
+            if model.id == id then
+                case kind of
+                    Message.Edit ->
+                        case model.state of
+                            Viewing content ->
+                                ( { model | state = Editing { content = content, cached = content } }
+                                , [ Focus (toHtmlID (getTextAreaID model)) ]
+                                )
+
+                            _ ->
+                                ( model, [] )
+
+                    Message.Save ->
+                        case model.state of
+                            Editing state ->
+                                if state.content /= state.cached then
+                                    ( { model | state = Saving state }, [ updateState.saveComment state.content ] )
+
+                                else
+                                    ( { model | state = Viewing state.content }, [] )
+
+                            _ ->
+                                ( model, [] )
+
+            else
+                ( model, [] )
+
+        Message.EditCommentBar id content ->
+            if model.id == id then
+                case model.state of
+                    Editing state ->
+                        ( { model | state = Editing { state | content = content } }
+                        , [ SyncTextareaHeight (getTextAreaID model) ]
```

Comment:
One place where we're missing the `SyncTextareaHeight` calls is when we resize the window - not the end of the world, but would be nice